packages/gatsby-theme-docs/src/layouts/internals/sidebar.js

```diff
@@ -66,8 +60,77 @@ const LinkItem = styled.div`
   flex-direction: row;
   align-items: flex-end;
 `;
+const linkStyles = `
+  border-left: ${designSystem.dimensions.spacings.xs} solid ${designSystem.colors.light.surfaceSecondary1};
+  padding-left: calc(${designSystem.dimensions.spacings.m} - ${designSystem.dimensions.spacings.xs});
+  text-decoration: none;
+  color: ${designSystem.colors.light.textSecondary};
+  display: flex;
+  flex-direction: row;
+  align-items: flex-end;
+
+  :hover {
+    color: ${designSystem.colors.light.linkNavigation} !important;
+  }
+
+  > * + * {
+    margin: 0 0 0 ${designSystem.dimensions.spacings.s};
+  }
+`;
+const activeLinkStyles = `
+  border-left: ${designSystem.dimensions.spacings.xs} solid ${designSystem.colors.light.linkNavigation} !important;
+  color: ${designSystem.colors.light.linkNavigation} !important;
+`;
 
 const SidebarLink = (props) => {
+  const {
+    locationPath,
+    customStyles,
+    customActiveStyles,
+    ...forwardProps
+  } = props;
+  return (
+    <ClassNames>
+      {({ css: makeClassName }) => {
+        const linkClassName = makeClassName`
+          ${linkStyles}
+          ${customStyles}
+        `;
+        const activeClassName = makeClassName`
+          ${activeLinkStyles}
+          ${customActiveStyles}
+        `;
+        return (
+          <Link
+            {...forwardProps}
+            to={trimTrailingSlash(props.to)}
+            getProps={({ href }) => {
+              // Manually check that the link is the active one, even with trailing slashes.
+              // The gatsby link is by default configured to match the exact path, therefore we
+              // need to check this manually.
+              const linkPath = trimTrailingSlash(href);
+              if (linkPath === locationPath) {
+                return {
+                  className: [linkClassName, activeClassName].join(' '),
+                  'aria-current': 'page',
+                };
+              }
+              return { className: linkClassName };
+            }}
+          />
+        );
+      }}
+    </ClassNames>
+  );
+};
+SidebarLink.propTypes = {
+  to: PropTypes.string.isRequired,
+  locationPath: PropTypes.string,
+  customStyles: PropTypes.string,
+  customActiveStyles: PropTypes.string,
+};
+
+const SidebarLinkWrapper = (props) => {
```

Comment:
I split the `SidebarLink` component into two, with the wrapper containing the logic for the scrolling position.

The base link component is use for the release notes link.